File: java/src/main/java/com/gmail/fuskerr63/java/repository/LocationRepository.java
package com.gmail.fuskerr63.java.repository;

import com.gmail.fuskerr63.java.entity.ContactLocation;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

public interface LocationRepository {
    Single<List<ContactLocation>> getAll();
    Single<ContactLocation> getUserByContactId(int contactId);
    Completable insert(ContactLocation contactLocation);
    void delete(ContactLocation contactLocation);
}

File: java/src/main/java/com/gmail/fuskerr63/java/repository/LocationRepository.kt
package com.gmail.fuskerr63.java.repository

import com.gmail.fuskerr63.java.entity.ContactLocation
import io.reactivex.Completable
import io.reactivex.Single
import kotlinx.coroutines.flow.Flow

interface LocationRepository {
    fun getAll(): Single<List<ContactLocation>>
    fun getFlowUserById(contactId: Int): Flow<ContactLocation>
    fun getSingleUserById(contactId: Int): Single<ContactLocation>
    fun insert(contactLocation: ContactLocation): Completable
}